code/espurna/sensors/DallasSensor.h

```diff
@@ -120,37 +155,57 @@ class DallasSensor : public BaseSensor {
                 // Read scratchpads
                 for (unsigned char index=0; index<_devices.size(); index++) {
 
-                    // Read scratchpad
-                    if (_wire->reset() == 0) {
-                        // Force a CRC check error
-                        _devices[index].data[0] = _devices[index].data[0] + 1;
-                        return;
-                    }
+                    if (_devices[index].address[0] == DS_CHIP_DS2406) {
 
-                    _wire->select(_devices[index].address);
-                    _wire->write(DS_CMD_READ_SCRATCHPAD);
+                        uint8_t data[DS2406_STATE_BUF_LEN];
 
-                    uint8_t data[DS_DATA_SIZE];
-                    for (unsigned char i = 0; i < DS_DATA_SIZE; i++) {
-                        data[i] = _wire->read();
-                    }
+                        _wire->reset();
```

Comment:
I have noticed that the other sections does check for the 'expected' return value. Should it do the same here, and the one `reset()` below?

https://github.com/PaulStoffregen/OneWire/blob/master/OneWire.cpp#L164
> // Returns 1 if a device asserted a presence pulse, 0 otherwise.

Based on the commit history, hot-plugging is the reason for that branch:
90c51552c7358c0d0e86380f034256f865ec0a9b

Reply:
@mcspr you were right, sorry for that.
Fixed, thanks.